src/game/title_screen.c

```diff
@@ -1729,44 +1731,53 @@ static void Task_LensFlareAnim(void)
     };
 }
 
-#define TinyChaoGardenConfig ((u32 *)(EWRAM_START + 0x8))
-
+#include "../chao_garden/include/language.h"
+#include "../chao_garden/include/program_params.h"
 static void LoadTinyChaoGarden(void)
 {
     u32 chaoGardenLang;
-    u32 unk374 = gLoadedSaveGame->score;
+#if (GAME == GAME_SA2)
+    u32 score = gLoadedSaveGame->score;
 
     switch (gLoadedSaveGame->language) {
         case LANG_JAPANESE:
-            chaoGardenLang = 0;
+            chaoGardenLang = TCGLANG_JAPANESE;
             break;
         case LANG_GERMAN:
-            chaoGardenLang = 3;
+            chaoGardenLang = TCGLANG_GERMAN;
             break;
         case LANG_FRENCH:
-            chaoGardenLang = 2;
+            chaoGardenLang = TCGLANG_FRENCH;
             break;
         case LANG_SPANISH:
-            chaoGardenLang = 4;
+            chaoGardenLang = TCGLANG_SPANISH;
             break;
         case LANG_ENGLISH:
         case LANG_ITALIAN:
-            chaoGardenLang = 1;
+            chaoGardenLang = TCGLANG_ENGLISH;
             break;
         default:
             chaoGardenLang = gLoadedSaveGame->language & 1;
             break;
     }
+#endif
 
     gFlags |= FLAGS_8000;
+#if (GAME == GAME_SA1)
+    m4aSongNumStop(MUS_CHARACTER_SELECTION);
+#else
     m4aMPlayAllStop();
+#endif
     m4aSoundVSyncOff();
     LZ77UnCompWram(gMultiBootProgram_TinyChaoGarden, (void *)EWRAM_START);
 
-    // TODO: what is going on here, doesn't work as a struct
-    // TODO: what's unk374
-    TinyChaoGardenConfig[0] = unk374;
+#if (GAME == GAME_SA1)
```

Comment:
Tbh, are we really gonna share these files between sa1 and sa2? I would imagine the rest of the file is completely differnent.